components/download_images/src/main.py

```diff
@@ -141,46 +128,41 @@ def __init__(self,
             max_aspect_ratio=max_aspect_ratio,
         )
 
-    def transform(self, dataframe: dd.DataFrame) -> dd.DataFrame:
-
-        logger.info(f"Length of the dataframe: {len(dataframe)}")
-        logger.info("Downloading images...")
-
-        # drop width and height columns, as those are going to be
-        # added later on
-        dataframe = dataframe.drop(columns=['images_width', 'images_height'])
-
-        # create meta
-        # needs to be a dictionary with keys = column names, values = dtypes of columns
-        # for each column in the output
-        meta = dict(zip(dataframe.columns, dataframe.dtypes))
-        meta["data"] = bytes
-        meta["width"] = int
-        meta["height"] = int
-
-        dataframe = dataframe.map_partitions(
-            download_image_with_retry_partition,
-            timeout=self.timeout,
-            retries=self.retries,
-            resizer=self.resizer,
-            meta=meta,
-        )
+    def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
+
+        logger.info(f"Downloading {len(dataframe)} images...")
+
+        results: t.List[t.Tuple[str]] = []
+        loop = asyncio.new_event_loop()
 
-        # rename new columns to be conform the spec
-        logger.info("Renaming columns...")
-        dataframe = dataframe.rename(columns={"data": "images_data",
-                                              "width": "images_width",
-                                              "height":"images_height"})
+        async def async_download():
+            with concurrent.futures.ThreadPoolExecutor(max_workers=20) as executor:
```

Comment:
are we utilizing the maximumx capacity if we set `max_workers=20` should we set it to by based on the cores in case we go for one of the heavy node pools?

Reply:
This is per core. I haven't done any optimization on this yet.